src/clr-angular/layout/tabs/tab-content.ts

```diff
@@ -3,35 +3,44 @@
  * This software is released under MIT license.
  * The full license information can be found in LICENSE in the root directory of this project.
  */
-import { Component, Inject, Input } from '@angular/core';
+import { Component, Inject, Input, TemplateRef, ViewChild } from '@angular/core';
 import { IF_ACTIVE_ID, IfActiveService } from '../../utils/conditional/if-active.service';
 import { AriaService } from './providers/aria.service';
+import { TabsService } from './providers/tabs.service';
 
 let nbTabContentComponents: number = 0;
 
 @Component({
   selector: 'clr-tab-content',
   template: `
+    <ng-template #tabContentProjectedRef>
       <section [id]="tabContentId" role="tabpanel" class="tab-content" [class.active]="active"
                [hidden]="!active"
                [attr.aria-labelledby]="ariaLabelledBy"
                [attr.aria-expanded]="active"
                [attr.aria-hidden]="!active">
         <ng-content></ng-content>
       </section>
+    </ng-template>
     `,
 })
 export class ClrTabContent {
   constructor(
     public ifActiveService: IfActiveService,
     @Inject(IF_ACTIVE_ID) public id: number,
-    private ariaService: AriaService
+    private ariaService: AriaService,
+    private tabsService: TabsService
   ) {
     if (!this.tabContentId) {
       this.tabContentId = 'clr-tab-content-' + nbTabContentComponents++;
     }
   }
 
+  @ViewChild('tabContentProjectedRef')
+  private set templateRef(value: TemplateRef<ClrTabContent>) {
+    this.tabsService.tabContentViewContainer.createEmbeddedView(value);
```

Comment:
I think we are forgetting to clear the view when the tab content becomes inactive. So as a result, it will embed the same view on every time the tab content instantiates with `*clrIfActive`.

Reply:
I can't reproduce this. The reference gets overwritten on tab change with the new template reference. Can you provide more details? Or explain how do you imagine a test for this?
I don't mind a proper cleanup, but want to be sure I understand you well.

Reply:
In tabs with `*clrIfActive`, if you leave a tab and come back to it, you would see the content being appended over and over again. So the test should check we have only one embedded tab content no matter how many times we come back to the same tab.

Reply:
Fixed. And a test added. Good catch, Shijir!